### ISSUE TITLE
Description of how to preprocess text corpus for roBERTa LM training

### ISSUE THREAD

**Issue by PhilipMay** [pytorch/fairseq#2270]
## 🚀 Feature Request
I am talking about this notebook where you show how to train a roBERTa LM:
https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.pretraining.md

But nobody sais exactly how to preprocess the text corpus. I think it must be one sentence per row. But does it need empty lines between documents? Is it ok to shuffle the text line by line?

Could you please claify this?

**Comment by lematt1991** [comment]
> I think it must be one sentence per row.

It doesn't necessarily have to be 1 sentence per row.  You can put any amount of text on a line, but be aware of limitations your model may impose (ex: max 512 tokens).  

> But does it need empty lines between documents?

If you want to use `--sample-break-mode complete_doc`, then yes, otherwise it isn't necessary

> Is it ok to shuffle the text line by line?

You shouldn't need to shuffle, you can specify whether or not the data should be shuffled in the dataloader.

**Comment by PhilipMay** [comment]
Hi @lematt1991 thanks for the detailed answer.
So basicaly everything is possible. Could you maybe tell me what option is better for the language model downstream performance?
Which language model would probably be better?

1. The one with empty lines between documents or without?
2. The one with shuffled lines or without?
3. The one with clean splittet sentences or without?

To ask more specifically:
How much does it influence the model when there are no blank lines between documents, no clean sentence splitting and everything is shuffled?
